scripts/src/main/scala/com/gu/mediaservice/scripts/EsScript.scala

```diff
@@ -42,81 +49,132 @@ object MoveIndex extends EsScript {
 
 object Reindex extends EsScript {
 
-  def run(esHost: String, extraArgs: List[String]) {
-
+  def run(esHost: String, args: List[String]) = {
     object EsClient extends ElasticSearchClient {
       val imagesAlias = "writeAlias"
       val port = esPort
       val host = esHost
       val cluster = esCluster
+      val initTime = DateTime.now()
+      val currentIndex = getCurrentIndices.reverse.head
+      val nextIndex = nextIndexName(currentIndex)
 
-      def reindex {
-        val scrollTime = new TimeValue(10 * 60 * 1000) // 10 minutes in milliseconds
-        val scrollSize = 500
-        val srcIndex = getCurrentAlias.get // TODO: error handling if alias isn't attached
-
+      def nextIndexName(currentIndex: String) = {
         val srcIndexVersionCheck = """images_(\d+)""".r
-        val srcIndexVersion = srcIndex match {
+        val srcIndexVersion = currentIndex match {
           case srcIndexVersionCheck(version) => version.toInt
           case _ => 1
         }
-        val newIndex = s"${imagesIndexPrefix}_${srcIndexVersion+1}"
+        s"${imagesIndexPrefix}_${srcIndexVersion + 1}"
+      }
+    }
 
-        createIndex(newIndex)
+    def raise(msg: String) = {
+      System.err.println(s"Reindex error on: $esHost : $msg ")
+      System.err.println("Exiting...")
+      EsClient.client.close()
+      System.exit(1)
+    }
 
-        val query = client.prepareSearch(srcIndex)
-          .setTypes(imageType)
-          .setScroll(scrollTime)
-          .setQuery(matchAllQuery)
-          .setSize(scrollSize)
-          .addSort("uploadTime", SortOrder.ASC)
+    def validateCurrentState(esClient: ElasticSearchClient, from: Option[DateTime]) = {
+      if(esClient.getCurrentIndices.isEmpty) raise("no index with the 'write' alias exists")
+      if((esClient.getCurrentIndices.length == 2) && from.isEmpty)
+        raise("there are two indices with 'write' alias, check your properties file or use http://localhost:9200/_plugin/head/ ")
+    }
 
-        def reindexScroll(scroll: SearchResponse, done: Long = 0) {
+    val imageType = "image"
+    val scrollTime = new TimeValue(5 * 60 * 1000) // 5 minutes in milliseconds
+    val scrollSize = 500
+    val currentIndex = EsClient.currentIndex
+    val newIndex = EsClient.nextIndex
+
+    val from = if(args.isEmpty) None else Some(DateTime.parse(args(0)))
+    validateCurrentState(EsClient, from)
+    Await.result(reindex(from, EsClient), Duration.Inf)
+
+    def reindex(from: Option[DateTime], esClient: ElasticSearchClient) : Future[SearchResponse] = {
+      def _scroll(scroll: SearchResponse, done: Long = 0): Future[SearchResponse] = {
+        val client = esClient.client
+        val doing = done + scrollSize
```

Comment:
haha i'm not sure if you're referring to "doing" but i borrowed it from the old code... funny how things dont seem funny after looking at them for a month >.< how about "currentPage" or "currentBatch"